base58/__init__.py

```diff
@@ -74,9 +88,23 @@ def b58decode_int(
     v = v.rstrip()
     v = scrub_input(v)
 
+    # map = array.array('b', [alphabet.find(x) for x in range(256)])
+    # map = bytes(fix_position(alphabet.find(x)) for x in range(256))
+
+    map = GLOBAL_MAP.get(alphabet, None)
```

Comment:
rather than dealing with caching in decode lets use `functools.lru_cache` to abstract this bit away

```
@lru_cache
def get_map(alphabet: string):
    return bytes(...)
```

Reply:
cool! Did not know about lru_cache. This solves potential problem with too big cache in case somebody will try using numerous alphabets